### PR TITLE
Refactor bnsapi

### PR DESCRIPTION
**Pull request by orkunkl**
With this PR bnsapi is more organised thus it will be easier to maintain `weave-v0.21.x` branch

### CODE REVIEW

**Comment by orkunkl** [comment]
@antoineherzog Can you check if any endpoints missing?

File: cmd/bnsapi/docs/swagger.yaml [hunk 1]
```diff
@@ -287,13 +396,46 @@ paths:
...
+            $ref: '#/definitions/username.Token'
+        "404": {}
+        "500": {}
+      summary: Returns the username object with associated info for an owner
+      tags:
+      - Starname
+  /username/resolve/{username}:
+    get:
+      parameters:
+      - description: 'username. example: bob*iov'
```
Comment (davepuchyr) [inline cmd/bnsapi/docs/swagger.yaml]: Please use `thematrix*iov` or other existing username that has more than one target.

**Comment by davepuchyr** [comment]
@orkunkl, really nice enhancements to the docs.

When I try the suggested parameters for address as hex or bech32 the results are empty.  Can we choose examples in the swagger docs that produce interesting results like you did for `thematrix*iov`?

Also, when I try querying 

`/username/owner/04C3DB7CCCACF58EEFCC296FF7AD0F6DB7C2FA17`

swagger produces

`curl -X GET "http://localhost:8000/username/owner/{address}" -H "accept: application/json"`
and
`{
  "errors": [
    "wrong input, must be address"
  ]
}`


That was with me running bnsapi as `TENDERMINT=https://rpc-private-a-vip-mainnet.iov.one:443 bnsapi`, which produces this log message: 

`dev username_handler.go:29: cannot decode hex: encoding/hex: invalid byte: U+007B '{' [iov-one/weave@v1.0.0/errors/errors.go:228]`.

**Comment by orkunkl** [comment]
> @orkunkl, really nice enhancements to the docs.
> 
> When I try the suggested parameters for address as hex or bech32 the results are empty. Can we choose examples in the swagger docs that produce interesting results like you did for `thematrix*iov`?
> 
> Also, when I try querying
> 
> `/username/owner/04C3DB7CCCACF58EEFCC296FF7AD0F6DB7C2FA17`
> 
> swagger produces
> 
> `curl -X GET "http://localhost:8000/username/owner/{address}" -H "accept: application/json"`
> and
> `{ "errors": [ "wrong input, must be address" ] }`
> 
> That was with me running bnsapi as `TENDERMINT=https://rpc-private-a-vip-mainnet.iov.one:443 bnsapi`, which produces this log message:
> 
> `dev username_handler.go:29: cannot decode hex: encoding/hex: invalid byte: U+007B '{' [iov-one/weave@v1.0.0/errors/errors.go:228]`.

@davepuchyr Will be resolved by https://github.com/iov-one/bns/pull/44. Thanks!

**Pull request merged**